.werft/jobs/build/self-hosted-upgrade-tests.ts

```diff
@@ -80,10 +80,11 @@ export async function triggerUpgradeTests(werft: Werft, config: JobConfig, usern
 
 export async function triggerSelfHostedPreview(werft: Werft, config: JobConfig, username: string) {
     const replicatedChannel =  config.replicatedChannel || config.repository.branch;
+    const version =  config.replicatedVersion || "-"
     const cluster =  config.cluster || "k3s";
-    const formattedBranch = config.repository.branch.replace("/", "-").slice(0,11)
+    const formattedBranch = config.repository.branch.replace("/", "-").slice(0,10)
     const phase = phases[cluster]
-    const subdomain =  `${formattedBranch}-${phase.cloud}`
+    const subdomain =  `${formattedBranch}y-${phase.cloud}`
```

Comment:
This is a hack, since sometimes the `slice` command above produces strings that end with `-`, if we don't have this `y` here, it created a string with consecutive `-`(hyphen). This was not an accepted naming format for several resources including DNS name

Reply:
**polish(non-blocker):** Instead of adding a suffix to the formatted branch name, can we either trim off a trailing hyphen or do a regex search and replace to remove repeated hyphens (along the lines of  `formattedBranch.replaceAll("-+", "-")`)?